proposals/lip_define_state_and_state_transitions_of_dpos_module.md

```diff
@@ -0,0 +1,1160 @@
+
+* `address`:  A 20-byte value identifying the voter.
+
+
+##### Returns
+
+This functions returns `voterStore(address)` deserialized using `voterStoreSchema`.
+
+
+#### getDelegate
+
+Returns the stored information relative to the given address.
+
+
+##### Parameters
+
+* `address`:  A 20-byte value identifying the delegate.
+
+
+##### Returns
+
+This functions returns `delegateStore(address)` deserialized using `delegateStoreSchema`.
+
+
+### Endpoints for Off-Chain Services
+
+
+#### getVoter(address)
+
+Returns voter information for the given address.
+
+
+##### Parameters
+
+* `address`: A 20-byte value identifying the voter.
+
+
+##### Returns
+
+This function returns `voterStore(address)` deserialized using `voterStoreSchema`
+
+
+#### getDelegate(address)
+
+Returns delegate information for the given address.
+
+##### Parameters
+
+* `address`: A 20-byte value identifying the delegate.
+
+##### Returns
+
+This functions returns `delegateStore(address)` deserialized using `delegateStoreSchema`.
+
+
+#### getAllDelegates()
+
+Returns information of all delegates.
+
+
+##### Parameters
+
+This function has no input parameter.
+
+
+##### Returns
+
+This function returns all `delegateStore` items deserialized using `delegateStoreSchema`.
+
+
+## Backwards Compatibility
+
+This LIP defines a new store interface for the DPoS module, which in turn will become part of the state tree and will be authenticated by the state root. 
+As such, it will induce a hardfork.
+
+
+## Reference Implementation
+
+TBA
+
+[LIP-0022]: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0022.md
+[LIP-0023]: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0023.md
+[LIP-0023-unlockTrs]: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0023.md#new-unlock-transaction
+[LIP-0023-voteTrs]: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0023.md#new-vote-transaction-1
+[LIP-0023-delegateProductivity]: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0023.md#delegate-productivity-1
+[LIP-0023-unlockRationale]: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0023.md#explicit-unlock-mechanism
+[LIP-0003]: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0003.md
+[LIP-0024-applyingPOM]: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0024.md#applying-a-pom-transaction
+[LIP-0024-verifyPOM]: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0024.md#validity-of-a-pom-transaction
+[LIP-0024-rationale]: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0024.md#rationale
+[LIP-0034-bootstrapPeriod]: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0034.md#bootstrap-period
+
+[LIP-newBlockHeader]: https://research.lisk.com/t/new-block-header-and-block-asset-schema/293
+[LIP-incentivizeCertificateGeneration]: https://research.lisk.com/t/introduce-unlocking-condition-for-incentivizing-certificate-generation/300
+[LIP-validators]: https://research.lisk.com/t/introduce-validators-module/317
```

Comment:
This link can be updated to the LIP repo now.

Reply:
indeed done.